TESTS/mbed_drivers/unbuffered_Serial/gt_unbuffered_serial_test_env.cpp

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright (c) 2019, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if DEVICE_SERIAL
+
+#include <string>
+
+#include "drivers/UnbufferedSerial.h"
+
+
+/**
+ * Macros for setting console flow control.
+ */
+#define CONSOLE_FLOWCONTROL_RTS                                               1
+#define CONSOLE_FLOWCONTROL_CTS                                               2
+#define CONSOLE_FLOWCONTROL_RTSCTS                                            3
+#define mbed_console_concat_(x) CONSOLE_FLOWCONTROL_##x
+#define mbed_console_concat(x) mbed_console_concat_(x)
+#define CONSOLE_FLOWCONTROL mbed_console_concat(MBED_CONF_TARGET_CONSOLE_UART_FLOW_CONTROL)
+
+
+/**
+ *  \brief Current token of key-value protocol's tokenizer
+ */
+static int cur_token = 0;
+
+/**
+ *  \enum Token enumeration for key-value protocol tokenizer
+ *
+ *        This enum is used by key-value protocol tokenizer
+ *        to detect parts of protocol in stream.
+ *
+ *        tok_eof       ::= EOF (end of file)
+ *        tok_open      ::= "{{"
+ *        tok_close     ::= "}}"
+ *        tok_semicolon ::= ";"
+ *        tok_string    ::= [a-zA-Z0-9_-!@#$%^&*()]+    // See isstring() function
+ *
+ */
+enum Token {
+    tok_eof = -1,
+    tok_open = -2,
+    tok_close = -3,
+    tok_semicolon = -4,
+    tok_string = -5
+};
+
+
+class GreenteaUnbufferedSerial : public mbed::UnbufferedSerial {
```

Comment:
You don't need a new class if you are not adding any function.
Just declare the main object as
`UnbufferedSerial test_unbuffered_serial(USBTX, USBRX, MBED_CONF_PLATFORM_STDIO_BAUD_RATE);`

Reply:
The purpose of the class is similar to the purpose of the existing `GreenteaSerial` class in `mbed-os/features/frameworks/greentea-client/source/greentea_serial.cpp`, set flow control in the constructor.

Reply:
You are correct that it is done this way for GreenteaSerial but we should not repeat the mistake. There is no need to create a new class because `set_flow_control`  is public, not protected and there is no additonal member functions/variables.
This file is not needed based on my previous comments. `test_unbuffered_serial` can/should be declared in main.cpp and you can set flow control as part of test case setup.

Reply:
Please review [this](https://github.com/hugueskamba/mbed-os/compare/310a7cd73ede0bcfff6c99b4b3ae1c891754b989..dd9ce703013101cd3ddc946a7e0f393357532b49).